### PR TITLE
treewide: use vim instead of nano

### PR DESCRIPTION
**Pull request by MatthewCroughan**
## Description of changes

Instead of including `nano` by default, include and use `vim` (actually `nvim`) by default

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/config/system-path.nix [hunk 1]
```diff
@@ -153,7 +153,6 @@ in
         "/sbin"
         "/share/emacs"
         "/share/hunspell"
-        "/share/nano"
```
Comment (jopejoe1) [inline nixos/modules/config/system-path.nix:156]: This should be added to the nano config

**Comment by pluiedev** [comment]
Why is this move necessary, exactly? Don't get me wrong, I'm a Neovim user myself — but (n)vim isn't exactly the easiest thing to understand how to use for new users. Nano isn't *perfect*, but it is more intuitive for the inexperienced than modal controls, `:wq` and hjkl.

**Review by K900 — CHANGES_REQUESTED** [review]
Really?

**Comment by piegamesde** [comment]
Why not emacs then? Or the new and shiny helix?

I'd say that pushing for a modal editor in such a use case is a non-starter to me. If you dislike nano and want to provide a better default that's fine, there are many similar alternatives. (I personally use `micro` for that purpose for example and am very happy with it.) At the same time, please also vet the chosen alternative for binary and closure size—since it is going to be installed on *every single system* by default, it should not be bloated.

**Review by drupol — COMMENTED** [review]
Hello Matt',

Could you please elaborate on the rationale reasoning behind this?

I understand that the landscape of text editors is vast and diverse, with each user having their own personal preference. Regardless of our choice, it seems inevitable that a portion of our users may not be fully satisfied. So changing `nano` for `<anything>` is not something I would do, IMHO, it's pointless.

**Comment by eclairevoyant** [comment]
Not only does this leave existing users without a default editor for those who disabled neovim, but also breaks behavior for those who also have vi(m) on their system.

If having nano on your personal systems causes you concern, there's always
```nix
programs.nano.enable = false;
```

If you have a serious proposal to change the default editor, please make one that fits reasonable criteria mentioned above (non-modal editing, small closure size, self-explanatory keybind), and ensure to consider existing users, but I don't think we should be entertaining joke PRs.

**Comment by Aleksanaa** [comment]
At least nano tells its users how to exit nano.

**Comment by mweinelt** [comment]
> Besides, plenty of distros ship _both_ vim and nano by default — I don't quite understand why we couldn't just do that

The reason is that we are reluctant to increase the size of our installer images.

Our installation guide could tell users how to get their favorite editor via `nix-shell` (or comparable tools). Personally I roll with `nix-shell -p helix nil`, providing me with instant LSP support in new installs, which is refreshing.

**Comment by MatthewCroughan** [comment]
I'm surprised most didn't think this was a joke.

**Comment by Aleksanaa** [comment]
> I'm surprised most didn't think this was a joke.

I'll not be surprised to know there are even more jokes in these 5600+ PRs.

**Comment by MatthewCroughan** [comment]
I am defeated!

**Comment by mweinelt** [comment]
:stuck_out_tongue:

**Comment by piegamesde** [comment]
> I'm surprised most didn't think this was a joke.

You'd be surprised how many PRs which are absolute jokes are opened in all seriousness by their author …

**Pull request closed without merging**